ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/HeartbeatAgent.java

```diff
@@ -233,42 +211,50 @@ protected ScheduledExecutorService executor() {
      * @param windowEnd the end time in ms of the current window
      */
     private void processHeartbeats(final long windowStart, final long windowEnd) {
+
+      final Map<HostInfo, HostStatus> copyOnWrite = new HashMap<>(immutableHostsStatus.get());
```

Comment:
note that this doesn't do a deep copy, so andy's comment (https://github.com/confluentinc/ksql/pull/4398/files#r372567512) is extra important, otherwise even though the map is immutable the entries within the map would be mutable (e.g. L220 actually affects the source map too!)

Reply:
Yes, @AlanConfluent's PR and the subsequent merge of our both PR's adds the immutable internal objects. We wanted to do it one place to make merging easier but I guess it doesn't because it complicates reviewing. I am making all the internal ones immutable as well